src/Microsoft.Framework.MemoryPool/DefaultArraySegmentPool.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+
+namespace Microsoft.Framework.MemoryPool
+{
+    public class DefaultArraySegmentPool<T> : IArraySegmentPool<T>, IDisposable
+    {
+        public readonly static int Capacity = Environment.ProcessorCount * 4;
+        public readonly static int BlockSize = 4096;
+
+        private readonly ConcurrentQueue<DefaultLeasedArraySegment> _segments;
+        private volatile bool _isDisposed;
+
+        public DefaultArraySegmentPool()
```

Comment:
Enter key is broken.

I use powershell to insert `Environment.NewLine` everywhere I need it.